vm/interpreter/src/fvm.rs

```diff
@@ -198,28 +191,39 @@ impl<DB: Blockstore> Consensus for ForestExterns<DB> {
                 && bh_2.parents().cids().contains(bh_3.cid())
                 && !bh_2.parents().cids().contains(bh_1.cid())
             {
-                cf = Some(ConsensusFault {
-                    target: *bh_1.miner_address(),
-                    epoch: bh_2.epoch(),
-                    fault_type: ConsensusFaultType::ParentGrinding,
-                })
+                fault_type = Some(ConsensusFaultType::ParentGrinding);
             }
         };
 
         // (3) return if no consensus fault
-        if cf.is_none() {
-            return Ok((cf, total_gas));
+        if fault_type.is_none() {
+            return Ok((None, total_gas));
         }
 
         // (4) expensive final checks
 
         // check blocks are properly signed by their respective miner
         // note we do not need to check extra's: it is a parent to block b
         // which itself is signed, so it was willingly included by the miner
-        total_gas += self.verify_block_signature(&bh_1)?;
-        total_gas += self.verify_block_signature(&bh_2)?;
-
-        Ok((cf, total_gas))
+        if let Ok(gas_used) = self.verify_block_signature(&bh_1) {
+            total_gas += gas_used;
+            if let Ok(gas_used) = self.verify_block_signature(&bh_2) {
+                total_gas += gas_used;
+                let ret = Some(ConsensusFault {
+                    target: *bh_1.miner_address(),
+                    epoch: bh_2.epoch(),
+                    // Unwrapping is safe here, see (3)
+                    fault_type: fault_type.unwrap(),
```

Comment:
We could use a match instead of `fault_type.is_none()` to get rid of this `unwarp`.